src/lib.rs

```diff
@@ -606,25 +614,24 @@ impl<T: Counter> Histogram<T> {
         // NOT ok to be +/- 2 units at 1999. Only starting at 2000. So internally, we need to
         // maintain single unit resolution to 2x 10^decimal_points.
 
-        // largest value with single unit resolution
-        let largest = 2 * 10_u64.pow(sigfig);
+        // largest value with single unit resolution, in [2, 200_000].
+        let largest = 2 * 10_u32.pow(sigfig as u32);
 
-        let unit_magnitude = ((low as f64).log2() / 2_f64.log2()).floor() as usize;
+        let unit_magnitude = (low as f64).log2().floor() as u8;
         let unit_magnitude_mask = (1 << unit_magnitude) - 1;
 
         // We need to maintain power-of-two sub_bucket_count (for clean direct indexing) that is
         // large enough to provide unit resolution to at least
         // largest_value_with_single_unit_resolution. So figure out
-        // largest_value_with_single_unit_resolution's nearest power-of-two (rounded up), and use that:
-        let sub_bucket_count_magnitude = ((largest as f64).log2() / 2_f64.log2()).ceil() as usize;
-        let sub_bucket_half_count_magnitude = if sub_bucket_count_magnitude > 1 {
-            sub_bucket_count_magnitude
-        } else {
-            1
-        } - 1;
-        let sub_bucket_count = 2_usize.pow(sub_bucket_half_count_magnitude as u32 + 1);
-
-        if unit_magnitude + sub_bucket_half_count_magnitude + 1 > 63 {
+        // largest_value_with_single_unit_resolution's nearest power-of-two (rounded up), and use
+        // that.
+        // In [1, 18]. 2^18 > 2 * 10^5 (the largest possible
+        // largest_value_with_single_unit_resolution)
+        let sub_bucket_count_magnitude = (largest as f64).log2().ceil() as u8;
+        let sub_bucket_half_count_magnitude = sub_bucket_count_magnitude - 1;
+        let sub_bucket_count = 1_u32 << (sub_bucket_count_magnitude as u32);
```

Comment:
More half_count + 1 => count. Also, since `2^k == 1 << k`, this seems more consistent with all the other bitshifting going on.